Week3/homework/step2-6.js

```diff
@@ -4,18 +4,28 @@ const arr2d = [[1, 2], [3, 4], [5, 6]];
 const arr3d = [[[1, 2], [3, 4]], [[5, 6], [7, 8]]];
 
 function flattenArray2d(arr) {
-  // Replace this comment and the next line with your code
-  console.log(arr);
+  const result = arr.reduce((acc, x) => acc.concat(x));
+  return result;
 }
 
 function flattenArray3d(arr) {
-  // Replace this comment and the next line with your code
-  console.log(arr);
+  let result = arr;
+  for (let cnt = 0; cnt < 2; cnt++) {
```

Comment:
Here you are making an assumption that the array that is passed to you contains exactly two subarrays. In general, a function should not make such assumptions. For instance, your code will not produce a correct result if I pass an array with three subarrays:

```js
const arr3d = [[[1, 2], [3, 4]], [[5, 6], [7, 8]], [[9, 10], [11, 12]]];
```

Can you modify your code such that it will work with this array too (or any 3D array)?

Reply:
Does not matter "2" represent the dimension of the array. Your array also work in my code.

Reply:
Yes, you are completely right.